applications/nrf_desktop/doc/wheel.rst

```diff
@@ -1,7 +1,48 @@
 .. _wheel:
 
-Wheel module
+wheel module
 ############
 
-Wheel module is responsible for generating events related to the rotation of
+``wheel`` module is responsible for generating events related to the rotation of
 the mouse wheel.
+
+Configuration
+*************
+
+Module is enabled by :option:`CONFIG_DESKTOP_WHEEL_ENABLE`.
+
+For detecting rotation ``wheel`` uses Zephyr QDEC driver. Module can be enabled
+only when QDEC is configured in DTS (:option:`HAS_DTS_QDEC` is available).
+
+QDEC DTS configuration specifies how many steps are done during one full angle
+rotation. Sensor reports rotation data using angle.
+``wheel`` configuration option
+:option:`CONFIG_DESKTOP_WHEEL_SENSOR_VALUE_DIVIDER` is used to convert obtained
+value of angle into value passed to HID data subscriber.
+
+For example configuring QDEC with step 24 means that for each step sensor will
+report 15 degree of rotation. For HID to see this rotation as increment of
+one :option:`CONFIG_DESKTOP_WHEEL_SENSOR_VALUE_DIVIDER` should be set to 15.
+
+Implementation details
+**********************
+
+``wheel`` can be in following states:
+
+    * ``STATE_DISABLED``
+    * ``STATE_ACTIVE_IDLE``
+    * ``STATE_ACTIVE``
+    * ``STATE_SUSPENDED``
+
+When in ``STATE_ACTIVE`` ``wheel`` will enable QDEC driver and wait for callback
+indicating rotation. QDEC driver may consume power during its operation. If no
+rotation is detected after time specified in
+:option:``CONFIG_DESKTOP_WHEEL_SENSOR_IDLE_TIMEOUT`` ``wheel`` will switch
+to ``STATE_ACTIVE_IDLE`` in which QDEC is disabled. In this state motion is
+detected using GPIO interrupts. After motion is registered ``wheel`` switches
+back to ``STATE_ACTIVE``.
+
+When system enters low power state ``wheel`` goes to ``STATE_SUSPENDED``.
+In this state rotation is detected using GPIO interrupts. Module will issue
+a system wake up event when motion occurs.
```

Comment:
rotation occurrs (motion is related to motion_events)

Reply:
I wanted to use it as a synonym to not repeat rotation

Reply:
rephrased few things